rfcs/20230927-construct-tensor-without-user-handle/README.md

```diff
@@ -0,0 +1,103 @@
+# RFC: construct tensor without user handle
+
+## Introduction & Motivation
+
+In benchdnn and unit tests, users often need to allocate memory themselves. 
```

Comment:
You may add the current status of the API here, to make the introduction more clear.